app/ide-desktop/lib/dashboard/package.json

```diff
@@ -34,26 +34,29 @@
     "@fortawesome/fontawesome-svg-core": "^6.4.2",
     "@fortawesome/free-brands-svg-icons": "^6.4.2",
     "@fortawesome/react-fontawesome": "^0.2.0",
+    "@hookform/resolvers": "^3.4.0",
     "@monaco-editor/react": "4.6.0",
     "@sentry/react": "^7.74.0",
-    "@tanstack/react-query": "^5.27.5",
+    "@tanstack/react-query": "5.37.1",
```

Comment:
why do these few specific dependencies have pinned versions?

Reply:
Because dev tools pin the exact version of react-query and if don't pin the react-query here, we end up with duplicates in our node_modules. This leads to unexpected behavior (recently I ran into an issue when dev tools failed to resolve a client instance from context. The problem was in duplicated react-query - the context we were using was different from the context dev tools were using)

Reply:
Ideally, we should always pin versions of our dependencies to increase predictability (lock file doesn't always guarantee that, due to deduplication and hoisting issues). Ideally, we'd better migrate to pnpm because I've already had issues when vite failed to resolve a dependency because npm failed to hoist correctly.

Reply:
i think (?) we're trying to move to bazel + pnpm, not sure what progress is like there though.